FirebaseInAppMessaging/Swift/Source/CustomInAppMessageDisplayViewModifier.swift

```diff
@@ -0,0 +1,187 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseInAppMessaging
+import SwiftUI
+
+// MARK: Image-only messages.
+
+@available(iOS 13, tvOS 13, *)
+struct ImageOnlyInAppMessageDisplayViewModifier<DisplayMessage: View>: ViewModifier {
+  var closure: (InAppMessagingImageOnlyDisplay, InAppMessagingDisplayDelegate) -> DisplayMessage
+  @ObservedObject var delegateBridge: DelegateBridge = DelegateBridge.shared
+
+  func body(content: Content) -> some View {
+    return content.overlay(overlayView())
+  }
+
+  @ViewBuilder
+  func overlayView() -> some View {
+    if let imageOnlyMessage = delegateBridge.inAppMessageData?.0 as? InAppMessagingImageOnlyDisplay,
+      let delegate = delegateBridge.inAppMessageData?.1 {
+      closure(imageOnlyMessage, delegate)
+        .onAppear { delegate.impressionDetected?(for: imageOnlyMessage) }
+    } else {
+      EmptyView()
+    }
+  }
+}
+
+@available(iOS 13, tvOS 13, *)
+public extension View {
+  /// Overrides the default display of an image only in-app message as defined on the Firebase console.
+  func imageOnlyInAppMessage<Content: View>(closure: @escaping (InAppMessagingImageOnlyDisplay,
+                                                                InAppMessagingDisplayDelegate)
+      -> Content)
+    -> some View {
+    modifier(ImageOnlyInAppMessageDisplayViewModifier(closure: closure))
+  }
+}
+
+// MARK: Banner messages.
+
+@available(iOS 13, tvOS 13, *)
+struct BannerInAppMessageDisplayViewModifier<DisplayMessage: View>: ViewModifier {
+  var closure: (InAppMessagingBannerDisplay, InAppMessagingDisplayDelegate) -> DisplayMessage
+  @ObservedObject var delegateBridge: DelegateBridge = DelegateBridge.shared
```

Comment:
Might not need to explicitly declare the type, as it can be inferred.

Reply:
Removed in 34ba6ac.